### ISSUE TITLE
Add support for Fortran scope/block snippets

### ISSUE THREAD

**Issue by hansec** [hansec/fortran-language-server#47]
Add snippet support for common fortran blocks (`if`, `do`, etc.) and scope declarations (`module`, `subroutine`, etc.).

examples ($0 denotes cursor position)
```fortran
if($0)then
end if
```

```fortran
module $0()
end module $0
```

**Comment by PaulXiCao** [comment]
I found some other projects which have alrady implemented various snippets. Maybe we should just copy-paste them...

- [atom - language-fortran](https://atom.io/packages/language-fortran): snippet file [here](https://github.com/dparkins/language-fortran/blob/master/snippets/language-fortran.cson)
- [vscode - fortran](https://marketplace.visualstudio.com/items?itemName=Gimly81.fortran): snippet file [here](https://github.com/Gimly/vscode-fortran/blob/master/snippets/fortran.json)
- [vscode - modern fortran](https://marketplace.visualstudio.com/items?itemName=krvajalm.linter-gfortran): snippet file [here](https://github.com/krvajal/vscode-fortran-support/blob/master/snippets/fortran90.json)

**Comment by michaelkonecny** [comment]
Is there a way to make these a bit more intelligent?
In Atom, I often find myself typing `end sub`<kbd>tab</kbd> and that inserts the snipped for an entire new subroutine, which is super annoying.
Perhaps just adding a new snippet for `end subroutine` with prefix `end sub` would override the `subroutine` one in this case?

**Comment by hansec** [comment]
Yeah, one of the reasons I have not added this yet is that for Atom and VSCode at least the most common snippets are already provided (since the "language support" packages are required for the LS anyways). Beyond that getting the LS provided snippets and the "language support" package snippets to play nice is a little tricky, since you end up with two snippets that are almost identical. I hope to have some time to look in to this again soon though.
